src/state/slices/swappersSlice/swappersSlice.ts

```diff
@@ -34,13 +36,34 @@ export const swappers = createSlice({
     clear: () => initialState,
     setBuyAsset: (state, action: PayloadAction<Asset>) => {
       state.buyAsset = action.payload
+
+      const buyAssetChainId = state.buyAsset.chainId
+      const buyAssetAccountChainId = state.buyAssetAccountId
+        ? fromAccountId(state.buyAssetAccountId).chainId
+        : undefined
+
+      // reset user selection on mismatch
+      if (state.buyAssetAccountId && buyAssetChainId !== buyAssetAccountChainId)
+        state.buyAssetAccountId = undefined
     },
     setSellAsset: (state, action: PayloadAction<Asset>) => {
       state.sellAsset = action.payload
+
+      const sellAssetChainId = state.sellAsset.chainId
+      const sellAssetAccountChainId = state.sellAssetAccountId
+        ? fromAccountId(state.sellAssetAccountId).chainId
+        : undefined
+
+      // reset user selection on mismatch
+      if (state.sellAssetAccountId && sellAssetChainId !== sellAssetAccountChainId)
+        state.sellAssetAccountId = undefined
```

Comment:
this is the last piece of "high risk" logic remaining. Refactoring everything meant the actual high risk stuff is no longer required.